sklearn/tree/_tree.pyx

```diff
@@ -91,23 +91,33 @@ cdef class TreeBuilder:
         """Build a decision tree from the training set (X, y)."""
         pass
 
-    cdef inline _check_input(self, object X, cnp.ndarray y,
-                             cnp.ndarray sample_weight):
-        """Check input dtype, layout and format"""
+    cdef inline _check_input(
+        self,
+        object X,
+        cnp.ndarray y,
+        cnp.ndarray sample_weight
+    ):
+        """Check input dtype, layout and format."""
         if issparse(X):
+            # since sparse arrays are represented by three 1D arrays, they are all
+            # fortran by default
```

Comment:
If we need a comment here:

```suggestion
            # CSC matrices are naturally F-ordered
```

Aside: CSR matrices are also three 1d arrays, but they are "naturally C-ordered".